apollo-router/src/plugins/telemetry/apollo_otlp_exporter.rs

```diff
@@ -0,0 +1,140 @@
+use std::borrow::Cow;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+use derivative::Derivative;
+use futures::future::BoxFuture;
+use opentelemetry::sdk::export::trace::ExportResult;
+use opentelemetry::sdk::export::trace::SpanData;
+use opentelemetry::sdk::export::trace::SpanExporter;
+use opentelemetry::sdk::trace::EvictedQueue;
+use opentelemetry::sdk::Resource;
+use opentelemetry::trace::SpanContext;
+use opentelemetry::trace::Status;
+use opentelemetry::trace::TraceFlags;
+use opentelemetry::trace::TraceState;
+use opentelemetry::InstrumentationLibrary;
+use opentelemetry::KeyValue;
+use opentelemetry_otlp::SpanExporterBuilder;
+use opentelemetry_otlp::WithExportConfig;
+use sys_info::hostname;
+use tonic::metadata::MetadataMap;
+use tonic::metadata::MetadataValue;
+use tower::BoxError;
+use url::Url;
+use uuid::Uuid;
+
+use super::tracing::apollo_telemetry::LightSpanData;
+use crate::plugins::telemetry::apollo::ROUTER_ID;
+use crate::plugins::telemetry::apollo_exporter::get_uname;
+use crate::plugins::telemetry::tracing::BatchProcessorConfig;
+use crate::plugins::telemetry::GLOBAL_TRACER_NAME;
+
+/// The Apollo Otlp exporter is a thin wrapper around the OTLP SpanExporter.
+#[derive(Clone, Derivative)]
+#[derivative(Debug)]
+pub(crate) struct ApolloOtlpExporter {
+    batch_config: BatchProcessorConfig,
+    endpoint: Url,
+    apollo_key: String,
+    resource_template: Resource,
+    intrumentation_library: InstrumentationLibrary,
+    #[derivative(Debug = "ignore")]
+    otlp_exporter: Arc<Mutex<opentelemetry_otlp::SpanExporter>>,
```

Comment:
Could you use https://github.com/Amanieu/parking_lot mutex ? It's already in our deps and usually more perfs

Reply:
Done, seems to work fine so far.